static/js/src/cube/components/CubeHeader/Hero.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Col, Row } from "@canonical/react-components";
+import { Module } from "../../types";
+
+type Props = {
+  modules: Module[];
+  isLoading: boolean;
+  certifiedBadge: null | Record<string, string>;
+  hasEnrollments: boolean;
+};
+
+const Hero = ({
+  modules,
+  isLoading,
+  certifiedBadge,
+  hasEnrollments,
+}: Props) => {
+  const [heroStripIsShallow, setHeroStripIsShallow] = useState(false);
+  const [heroContentMarginTop, setHeroContentMarginTop] = useState("2rem");
+  const heroStrip = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        const width = Math.floor(entry.contentRect.width);
+        if (width < 1035) {
+          setHeroStripIsShallow(false);
+          setHeroContentMarginTop("2rem");
+        } else {
+          setHeroStripIsShallow(true);
+          setHeroContentMarginTop("-2rem");
+        }
+      });
+    });
+
+    if (heroStrip.current) {
+      resizeObserver.observe(heroStrip.current);
+    }
+  }, [heroStrip.current]);
+
+  const passedCourses = modules.reduce(
+    (passedCount, module) => passedCount + (module.status === "passed" ? 1 : 0),
+    0
+  );
+
+  const heading = isLoading
+    ? ""
+    : certifiedBadge
+    ? "CUBE 2020 complete"
+    : hasEnrollments
+    ? "Your road to CUBE"
+    : "The road to CUBE";
+
+  const description = isLoading
+    ? ""
+    : certifiedBadge
+    ? `You have completed all ${modules.length} microcerts, proving your Ubuntu expertise. You are now a Certified Ubuntu Engineer.`
+    : hasEnrollments
+    ? passedCourses > 0
+      ? `You have completed ${passedCourses} microcerts.\nComplete ${
+          modules.length - passedCourses
+        } more microcerts to attain CUBE.`
+      : `Complete ${modules.length} microcerts to attain CUBE.`
+    : `The path to certification is clear and convenient. Complete ${modules.length} microcerts to attain CUBE.`;
```

Comment:
same as above.

One idea could be to define heading and descriptions as empty variables and then do something like

```jsx
if(certifiedBadge){
  description = "..."
  heading = "Cube 2020 complete"
} else if(hasEnrollments){
  heading = "Your road to CUBE"
 ...
} else {
 ...
}
```

what do you think?

Reply:
Fixed :+1: